### PR TITLE
update the mpas_2024052700 case to use mpasout files

### PR DESCRIPTION
**Pull request by guoqing-noaa**
Use `mem001` instead of `mem01` to be consistent with the workflow

Address issue #160

### CODE REVIEW

**Comment by guoqing-noaa** [comment]
@Junjun-NOAA @delippi 
If you are interested, you may try rerunning your case using the mpasout files in this PR to see whether you can get wind increments from the assimilation of T obs.

**Comment by delippi** [comment]
@guoqing-noaa, yes, I already started. I was able to assimilate winds and get temperature increments (slightly different from what others are doing). I will have some suggestions in a review that I will work on tomorrow.

File: rrfs-test/scripts/setup_experiment.sh [hunk 1]
```diff
@@ -98,8 +98,8 @@ elif [[ $DYCORE == "MPAS" ]]; then
   mkdir -p data; cd data
   mkdir -p bumploc bkg obs ens
   ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/bumploc/* bumploc/.
-  ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/bkg/restart.2024-05-27_00.00.00.nc .
-  ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/bkg/restart.2024-05-27_00.00.00.nc static.nc
+  ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/bkg/mpasout.2024-05-27_00.00.00.nc .
+  ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/bkg/static.nc static.nc
```
Comment (delippi) [inline rrfs-test/scripts/setup_experiment.sh:102]: `ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/bkg/static.nc static.nc` should be `ln -snf ${YOUR_PATH_TO_RDASAPP}/fix/expr_data/${TEST_DATA}/static.nc static.nc` (the `bkg` removed)

**Review by delippi — CHANGES_REQUESTED** [review]
I found one small change that is needed. In `setup_experiment.sh` the path to `static.nc` is slightly incorrect. I noted what it should be in that file. Other than that this PR looks good to me. I did run everything using the new backgrounds and I can confirm that I was able to assimilate wind observations and was able to have increments of temperature. I was going to share my results, because I thought they were strange, but I just realized there might be issues with `data/obs`.

Edit: Actually, I think there are just issues with `data/obs/ioda_mesonet.nc` which I didn't test (due to incorrect data types and some other things--perhaps an old yaml was used to create that?). The aircraft data is correct.

File: rrfs-test/scripts/link_mpasjedi_expr.sh [hunk 2]
```diff
@@ -44,10 +44,10 @@ cp ${RDASApp}/rrfs-test/ush/mpasjedi_spread.py .
 
 mkdir -p data
 cd data
-mkdir -p bumploc bkg obs ens ref
+mkdir -p bumploc obs ens ref
 ln -snf ${RDASApp}/fix/bumploc/${BUMPLOC} bumploc/${BUMPLOC}
-ln -snf ${RDASApp}/fix/expr_data/${exprname}/bkg/restart.2024-05-27_00.00.00.nc .
-ln -snf ${RDASApp}/fix/expr_data/${exprname}/bkg/restart.2024-05-27_00.00.00.nc static.nc
+ln -snf ${RDASApp}/fix/expr_data/${exprname}/bkg/mpasout.2024-05-27_00.00.00.nc .
+ln -snf ${RDASApp}/fix/expr_data/${exprname}/static.nc static.nc
```
Comment (TingLei-NOAA) [inline rrfs-test/scripts/link_mpasjedi_expr.sh:50]: could those static files in the source location be added  qualifiers of  grid-related labels?

**Comment by HuiLiu-NOAA** [comment]
@guoqing-noaa : just tested the new case and got an error with the files:

17: !!! Hint: group interp_c1b_to_c0a_002
 17:        |-> cmp_1
 17:           |-> common
 17:              |-> bumploc_401km11levels_nicas
 17: Abort(1) on node 17 (rank 17 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 17
 18: !!! NetCDF error in netcdf_inquire_grp on task #000019: NetCDF: No group found.

**Comment by ShunLiu-NOAA** [comment]
@guoqing-noaa Do you have a chance to address @HuiLiu-NOAA and @TingLei-NOAA's comments?

**Comment by guoqing-noaa** [comment]
@ShunLiu-NOAA Thanks for the reminder. I will address Hui and Ting's comments.

But the big concern is, as discussed with @SamuelDegelia-NOAA in this thread, the transition to mpasout files changed the bumploc files. This is unexpected and we will need more time to dig into this issue.

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa  Could you point to your run dirs to create bump files using restart files (old) and mpasout files (current) respectively?  This will allow us to reproduce your results and join the investigation into this issue.

**Comment by guoqing-noaa** [comment]
> @guoqing-noaa Could you point to your run dirs to create bump files using restart files (old) and mpasout files (current) respectively? This will allow us to reproduce your results and join the investigation into this issue.

@TingLei-NOAA Thanks for helping with this issue.

I ran the mpasout version on Jet and it is at:
`/lfs5/BMC/wrfruc/gge/RDASApp/expr/mpas_2024052700`   

I ran the restart version on Hera and it is at:
`/scratch1/BMC/wrfruc/gge/RDASApp/expr/mpas_2024052700`  

If you want to reproduce the case comparison,  you can check two versions of RDASApp: NOAA-EMC:develop and guoqing-noaa:mpasout, generate bumploc files respectively, run the single obs jedi case, plot the increments.

I can also regenerate them at a later time (maybe weekend).

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa   Thanks. I had copied them to my account and will see what the differences are. 
An update:  Guqing, would you point me to the bumploc.yaml used in your hera run? seems it is missing in the dir you pointed.

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa  My runs showed identical results using mpasout or restart files. 
What I did is as follows:
```
I changed the bumploc.yaml for jet case to be used for hera case with another variables changes which I believe shouldn't affect results (but we could check on it later if needed) . 
The two files are in https://github.com/TingLei-NOAA/RDASApp/tree/feature/for-test/rrfs-test/example-files. 
I ran both cases on hera using the executable: mpasjedi_error_covariance_toolbox.x from https://github.com/NOAA-EMC/RDASApp/pull/175.  I used it because its executables are already ready on hera.   
Using "diff" to compare the generated "bumploc_401km11levels_nicas_local*.nc" from the two runs and no differences were found. 
```

This confirms our pre-assumption that using mpasout or restart files should not change the output in the nicas files (while, there could be still other possibility like the differences in the vertical coor. to cause differences), which is , at least, understandable. 

So, the differences you found might come from other differences.  For further investigation, you could run those two cases also on the same machine , like hera. 
(my copy of your jet case on hera is /scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-guoqing/dr-jet/mpas_2024052700).
 
 We could first see you could get the same comparison results when running them on the same machine.

**Comment by ShunLiu-NOAA** [comment]
> @guoqing-noaa My runs showed identical results using mpasout or restart files. What I did is as follows:
> 
> ```
> I changed the bumploc.yaml for jet case to be used for hera case with another variables changes which I believe shouldn't affect results (but we could check on it later if needed) . 
> The two files are in https://github.com/TingLei-NOAA/RDASApp/tree/feature/for-test/rrfs-test/example-files. 
> I ran both cases on hera using the executable: mpasjedi_error_covariance_toolbox.x from https://github.com/NOAA-EMC/RDASApp/pull/175.  I used it because its executables are already ready on hera.   
> Using "diff" to compare the generated "bumploc_401km11levels_nicas_local*.nc" from the two runs and no differences were found. 
> ```
> 
> This confirms our pre-assumption that using mpasout or restart files should not change the output in the nicas files (while, there could be still other possibility like the differences in the vertical coor. to cause differences), which is , at least, understandable.
> 
> So, the differences you found might come from other differences. For further investigation, you could run those two cases also on the same machine , like hera. (my copy of your jet case on hera is /scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-guoqing/dr-jet/mpas_2024052700).
> 
> We could first see you could get the same comparison results when running them on the same machine.

@guoqing-noaa do you get a chance to take a look at this?

**Comment by guoqing-noaa** [comment]
Convert this to draft PR as there are lingering issues and we may update to the 66 levels directly and skip this PR.

**Comment by SamuelDegelia-NOAA** [comment]
Does #194 supersede this PR? Should we close this?

**Comment by guoqing-noaa** [comment]
> Does #194 supersede this PR? Should we close this?

Yes and good suggestion, @SamuelDegelia-NOAA 

@ShunLiu-NOAA, I quoted @TingLei-NOAA's replies below and thanks @TingLei-NOAA for the testing!

Yes, as Ting suggested, the differences came from the RDASApp hash I used is different from that was used to generate the staged bumploc files.  The `mpasout` and `restart` files themselves will not cause any differences in the bumploc files.

> @guoqing-noaa My runs showed identical results using mpasout or restart files. What I did is as follows:
> 
> ```
> I changed the bumploc.yaml for jet case to be used for hera case with another variables changes which I believe shouldn't affect results (but we could check on it later if needed) . 
> The two files are in https://github.com/TingLei-NOAA/RDASApp/tree/feature/for-test/rrfs-test/example-files. 
> I ran both cases on hera using the executable: mpasjedi_error_covariance_toolbox.x from https://github.com/NOAA-EMC/RDASApp/pull/175.  I used it because its executables are already ready on hera.   
> Using "diff" to compare the generated "bumploc_401km11levels_nicas_local*.nc" from the two runs and no differences were found. 
> ```
> 
> This confirms our pre-assumption that using mpasout or restart files should not change the output in the nicas files (while, there could be still other possibility like the differences in the vertical coor. to cause differences), which is , at least, understandable.
> 
> So, the differences you found might come from other differences. For further investigation, you could run those two cases also on the same machine , like hera. (my copy of your jet case on hera is /scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-guoqing/dr-jet/mpas_2024052700).
> 
> We could first see you could get the same comparison results when running them on the same machine.

**Comment by ShunLiu-NOAA** [comment]
Yes. We should close this one. @guoqing-noaa I did mention in this PR that we may consider keeping the restart file name.

**Comment by guoqing-noaa** [comment]
Close this PR as discussed. Thanks!

**Pull request closed without merging**